packages/gasket-plugin-metadata/lib/actions.js

```diff
@@ -28,23 +29,43 @@ module.exports = function actions(gasket) {
     getMetadata: async function getMetadata() {
       const app = getAppInfo(gasket);
       const plugins = [];
-      const modules = [];
       const presets = [];
+      const modules = {};
 
       await gasket.execApply('metadata', async (data, handler) => {
-        const pluginData = await handler(data);
-        pluginData.path = path.dirname(path.join(require.resolve(pluginData.name), '..'));
-        plugins.push(pluginData);
-
-        if (pluginData.modules) {
-          const moduleData = pluginData.modules.map(m => {
-            return { ...m, path: path.dirname(path.join(require.resolve(m.name), '..')) };
-          });
-          modules.push(...moduleData);
+        const isGasketPreset = data.name.startsWith('@gasket/preset-');
+        const isGasketPlugin = data.name.startsWith('@gasket/plugin-');
+        const isGasketPackage = isGasketModule.test(data.name) || isGasketPlugin || isGasketPreset;
+
+        if (!isGasketPackage) {
+          const pluginData = await handler(data);
+          pluginData.path = path.join(app.path, 'plugins');
+          plugins.push(pluginData);
+        } else {
+          const pluginData = await handler(data);
+          pluginData.path = path.dirname(path.join(require.resolve(pluginData.name), '..'));
+          const { dependencies, devDependencies } = require(path.join(pluginData.path, 'package.json'));
+
+          if (isGasketPreset)
+            presets.push(pluginData);
```

Comment:
Would **@godaddy/preset-**  work properly here?

Reply:
It wouldn't however I'm not sure we should pollute the OS scope with `@godaddy`? This is a good point and I'm not sure.